tianshou/policy/modelfree/td3.py

```diff
@@ -119,16 +118,22 @@ def learn(self, batch: Batch, **kwargs) -> Dict[str, float]:
         # critic 1
         current_q1 = self.critic1(batch.obs, batch.act).flatten()
         target_q = batch.returns.flatten()
-        critic1_loss = F.mse_loss(current_q1, target_q)
+        td1 = current_q1 - target_q
+        critic1_loss = (td1.pow(2) * batch.weight).mean()
+        # critic1_loss = F.mse_loss(current_q1, target_q)
```

Comment:
Make sure ``[n] * [n, 1] ==> [n, n]`` does not happen again :) The same for several occurrence above.

Reply:
Before this line, we ensure `current_q` and `target_q` are both 1-d array (same as batch.weight), so that cannot happen.